### PR TITLE
[MI-2232] Created an API to get states for a record type

### PR DESCRIPTION
**Pull request by manojmalik20**
Created route constants for both the ServiceNow API and the new plugin API Created serializer for ServiceNow state and the client funciton to get states for a record Created API to get the states for a record type

### CODE REVIEW

File: server/plugin/client.go [hunk 1]
```diff
@@ -219,3 +220,15 @@ func (c *client) AddComment(recordType, recordID string, payload *serializer.Ser
 	_, statusCode, err := c.CallJSON(http.MethodPatch, fmt.Sprintf("%s/%s", url, recordID), payload, nil, nil)
 	return statusCode, err
 }
+
+func (c *client) GetStatesFromServiceNow(recordType string) ([]*serializer.ServiceNowState, int, error) {
+	states := &serializer.ServiceNowStatesResult{}
+
```
Comment (ayusht2810) [inline server/plugin/client.go:226]:
```suggestion
```

**Review by ayusht2810 — APPROVED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/plugin/api.go [hunk 2]
```diff
@@ -636,6 +637,40 @@ func (p *Plugin) addCommentsOnRecord(w http.ResponseWriter, r *http.Request) {
...
+		p.handleAPIError(w, &serializer.APIErrorResponse{StatusCode: statusCode, Message: fmt.Sprintf("Error in getting the states. Error: %s", err.Error())})
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(statusCode)
+	result, err := json.Marshal(states)
+	if err != nil {
+		p.API.LogDebug("Error while marshaling the response", "Error", err.Error())
+		_, _ = w.Write([]byte("[]"))
```
Comment (Kshitij-Katiyar) [inline server/plugin/api.go:662]: do we need to check error for this ?

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/plugin/api.go [hunk 3]
```diff
@@ -636,6 +637,40 @@ func (p *Plugin) addCommentsOnRecord(w http.ResponseWriter, r *http.Request) {
...
+
+	client := p.GetClientFromRequest(r)
+	states, statusCode, err := client.GetStatesFromServiceNow(recordType)
+	if err != nil {
+		p.API.LogError("Error in getting the states", "Record Type", recordType, "Error", err.Error())
+		p.handleAPIError(w, &serializer.APIErrorResponse{StatusCode: statusCode, Message: fmt.Sprintf("Error in getting the states. Error: %s", err.Error())})
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
```
Comment (avas27JTG) [inline server/plugin/api.go:657]: have not we created a reusable kind of function for writing json content?

File: server/plugin/api.go (hunk 2 above)
Comment (manojmalik20) [reply server/plugin/api.go:662]: No, because I don't think it can fail ever.

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**